src/reducers/vms.js

```diff
@@ -254,14 +245,22 @@ const vms = actionReducer(initialState, {
     return state
   },
 
-  [SET_PAGE] (state, { payload: { page } }) {
-    return state.set('page', page)
-  },
-  [SET_CHANGED] (state, { payload: { value } }) {
-    return state.set('notAllPagesLoaded', value)
+  [UPDATE_PAGING_DATA] (state, { payload: { vmsPage, vmsExpectMorePages, poolsPage, poolsExpectMorePages } }) {
```

Comment:
how about moving this functionality to UPDATE_VMS reducer? This provide synchronization with the actual vm list.
This might help to solve some corner cases after refresh.

Reply:
`UPDATE_VMS` doesn't know, and doesn't need to know anything about paging.  It just needs to take the VMs and merge them into redux state.

> This might help to solve some corner cases after refresh.

What do you mean "after refresh"?

Reply:
After updating vms we should also check if paging data needs to be updated.  So both actions seem connected to me.

In `refreshListPage()` this is handled - thanks for clarification.